widgets/wifiiw.lua

```diff
@@ -0,0 +1,89 @@
+---------------------------------------------------
+-- Licensed under the GNU General Public License v2
+--  * (c) 2010, Adrian C. <anrxc@sysphere.org>
+---------------------------------------------------
+
+-- {{{ Grab environment
+local tonumber = tonumber
+local math = { ceil = math.ceil }
+local setmetatable = setmetatable
+local helpers = require("vicious.helpers")
+local io = {
+    open  = io.open,
+    popen = io.popen
+}
+local string = {
+    find  = string.find,
+    match = string.match
+}
+-- }}}
+
+
+-- Wifiiw: provides wireless information for a requested interface using iw instead of deprecated iwconfig
+-- vicious.widgets.wifiiw
+local wifiiw = {}
+
+
+-- {{{ Variable definitions
+local iw = "iw"
+local iwcpaths = { "/sbin", "/usr/sbin", "/usr/local/sbin", "/usr/bin" }
+-- }}}
+
+
+-- {{{ Wireless widget type
+local function worker(format, warg)
+    if not warg then return end
+
+    -- Default values
+    local winfo = {
+        ["{ssid}"] = "N/A",
+        ["{mode}"] = "N/A",
+        ["{chan}"] = 0,
+        ["{rate}"] = 0,
+        ["{freq}"] = 0,
+		["{txpw}"] = 0,
+        ["{sign}"] = 0
+    }
+
+    -- Sbin paths aren't in user PATH, search for the binary
+    if iw == "iw" then
+        for _, p in ipairs(iwcpaths) do
+            local f = io.open(p .. "/iw", "rb")
+            if f then
+                iw = p .. "/iw"
+                f:close()
+                break
+            end
+        end
+    end
+
+    -- Get data from iw where available
+    local f = io.popen(iw .." dev ".. warg .. " link 2>&1;" .. iw .." dev ".. warg .. " info 2>&1")
```

Comment:
actually path search could be also delegated to the shell, which does also check if the file is executable.

```
io.popen("export PATH=$PATH:/sbin/:/usr/sbin:/usr/local/sbin; iw dev " ...
```

In that case the loop above, can be removed.

Also please escape shell characters from warg using `helpers.shellquote(warg)`.